src/utils/settings.ts

```diff
@@ -2,25 +2,36 @@ import fs from 'fs'
 import { homedir } from 'os'
 import { join } from 'path'
 import { mergeDeepRight } from 'ramda'
+import yaml from 'js-yaml'
 
 import { createLogger } from '../factories/logger-factory'
-import defaultSettingsJson from '../../resources/default-settings.json'
 import { ISettings } from '../@types/settings'
 
 const debug = createLogger('settings')
 
+const FileType = {
+  yaml: 'yaml',
+  json: 'json',
+}
+
 export class SettingsStatic {
   static _settings: ISettings
 
-  public static getSettingsFilePath(filename = 'settings.json') {
-    return join(
-      process.env.NOSTR_CONFIG_DIR ?? join(homedir(), '.nostr'),
-      filename
-    )
+  public static getSettingsFileBasePath(): string {
+    return process.env.NOSTR_CONFIG_DIR ?? join(homedir(), '.nostr')
```

Comment:
bug here, this should not be `homedir()` anymore
it should be `process.cwd()`

Reply:
Is this right? You want the settings file to be inside of the `nostream/src/utils/.nostr` directory, not `~/.nostr`? Just making sure I understand correctly.